Safer defaults for immutable znc config

layus opened the pull request:
I just lost all the options I configured in ZNC, because the mutable config was overwritten.
I accept any suggestions on the way to implement this, but overwriting a mutable config by default seems weird. If we want to do this, we should ensure that ZNC does not allow to edit the config via the webmin when cfg.mutable is false.

layus committed:
Safer defaults for immutable znc config

I just lost all the options I configured in ZNC, because the mutable config was overwritten.
I accept any suggestions on the way to implement this, but overwriting a mutable config by default seems weird. If we want to do this, we should ensure that ZNC does not allow to edit the config via the webmin when cfg.mutable is false.

Mic92 commented on nixos/modules/services/networking/znc.nix:
```diff
@@ -379,7 +379,8 @@ in
           ${pkgs.coreutils}/bin/echo "znc is set to be system-managed. Now deleting old znc.conf file to be regenerated."
-          ${pkgs.coreutils}/bin/rm -f ${cfg.dataDir}/configs/znc.conf
+          ${pkgs.coreutils}/bin/echo "You can still recover you old config from '${cfg.dataDir}/configs/znc.conf.bak'."
+          ${pkgs.coreutils}/bin/mv ${cfg.dataDir}/configs/znc.conf{,.bak}
```
I would only move it, if it is not a symlink to the nix store. Otherwise a second restart will overwrite you existing configuration. This also avoids moving if no mutable configuration has ever existed.

layus replied:
It is never a symlink. If the file is not writeable, znc fails at startup. When it is not mutable, the file is overwritten at startup from the nix store file. Hence the issue.

Znc does not support immutable configuration files.

layus replied:
We could copy it only if it differs from the reference store file.

Mic92 replied:
If it is overwritten on every startup anyway, we could also patch znc to stop complaining about read-only configuration.

layus replied:
Patching ZNC seems the sane way to go, wit an error message when trying to change settings. I do not see how a patter would help, as probably ZNC overwrites the config on exit and on config change.

Mic92 replied:
Probably.

layus replied:
Well, not going to happen soon... https://github.com/znc/znc/issues/975

I now think this is the safest way to go. Disabling immutability by default seems enough. This backup thing would have helped for me but is not that important if mutable is an opt-out.

layus committed:
Do not backup old config files.

There seems to be little need for backups if mutable becomes a voluntary opt-out.

layus committed:
fixup

layus commented:
I think this is quite simple now, but changing the default may surprise old users, is there some way to warn them ?

Mic92 commented:
An entry in release-notes can help.

Mic92 commented:
That would be here:

```
nixos/doc/manual/release-notes/rl-1803.xml
```
